remote-build-coordinator/src/main/java/org/jboss/pnc/remotecoordinator/builder/BuildTasksInitializer.java

```diff
@@ -366,11 +361,18 @@ public static Collection<RemoteBuildTask> removeNRRTasks(Graph<RemoteBuildTask>
             }
         }
 
-        notToBuild.forEach(
-                task -> {
-                    // NOTE: after removal NRR task can still be referenced as a dependency of other tasks
-                    buildGraph.removeVertex(task);
-                });
+        notToBuild.forEach(task -> {
+            // Remove the reference if the task is not running (not present in Rex).
+            // If it is running, it will be linked by the Rex to a running dependency.
+            List<Edge<RemoteBuildTask>> dependantsEdges = task.getIncomingEdges();
```

Comment:
Extending the comment, does it make sense?
// Tasks that are not present in the Rex cannot be submitted to Rex,
// because Rex cannot reference them.

Reply:
You are right, there should be no case for a running NRR task.